kivy/uix/codeinput.py

```diff
@@ -129,7 +129,7 @@ def _create_line_label(self, text, hint=False):
             ntext = u'*' * len(ntext)
         ntext = self._get_bbcode(ntext)
         kw = self._get_line_options()
-        cid = u'{}\0{}\0{}'.format(ntext, self.password, kw)
+        cid = u'{}\0{}\0{}'.format(text, self.password, kw)
```

Comment:
What about the `ntext` variable and all the `bbcode` part?
Is something that we would like to change?

Reply:
Seems that everything else works fine, in particular without using cache (line 133).
The problem is that different `text`s can give the same `ntext`. At the same time `text` also used (line 141) after setting `ntext` (in contrast to TextInput._create_line_label).

Reply:
Tested. And now makes more sense to me. Thank you!